tests/python/gaia-ui-tests/gaiatest/apps/camera/app.py

```diff
@@ -59,6 +81,21 @@ def take_photo(self):
         # Wait for thumbnail to appear
         self.wait_for_thumbnail_visible()
 
+    def take_photo_with_timer(self):
+        # Wait for camera to be ready to take a picture
+        controls = self.marionette.find_element(*self._controls_locator)
+        Wait(self.marionette, timeout=20).until(
+            lambda m: controls.get_attribute('data-enabled') == 'true')
+         
+        self.tap_capture()
+        time.sleep(1)
+        # Wait for Timer to animate
+        timer_animating = self.marionette.find_element(*self._timer_animation_locator)
```

Comment:
when I run the script, it fails here, and it looks like it's because the tap_capture() didn't run properly.  (it didn't even tap the capture button)  I think we need more checks inside tap_capture() method.  How about we add a check there to see whether the capture button is visible?  I think in some cases marionette does not complain when you tap an invisible element.